### DISCUSSION TITLE
Single item matrix is excluded but still runs job

### DISCUSSION THREAD

**Discussion by Andrew-Chen-Wang** [community/community#26647] [Actions]
<p>I have a GitHub action job that has a single value in the matrix sometimes (since the matrix’s content is dynamic), but I want to exclude a certain item. However, if it’s a single item matrix (like one parameter and the array only has one item), how would I skip the job? Currently, the job keeps running, but it shouldn’t.</p>
<p>Here’s the workflow:</p>
<pre><code class="lang-auto">  deploy:
    name: Deploy
    runs-on: ubuntu-latest
    needs: [changes]
    if: ${{ needs.changes.outputs.packages != '[]' }}
    strategy:
      matrix:
        package: ${{ fromJSON(needs.changes.outputs.packages) }}
        exclude:
          - package: 'something'
</code></pre>
<p>So what eventually happens is <code>needs.changes.outputs.packages</code> is a single-element array like <code>["something"]</code> but something is excluded. However, the job still executes. The current if statement for the job protects against a completely empty matrix, but is there a way to catch this case of exclusion and prevent the job from running?</p>

**Comment by mike-kaminski** [comment, marked as answer]
<p>You should be able to extend the <code>if:</code> to something like this: <code>${{ (needs.changes.outputs.packages != '[]' &amp;&amp; needs.changes.outputs.packages != '[something]') }}</code>, or fix up the changes job to omit the ‘something’ package from the output, either up front before the changes output value is populated (detected changes to an env var with a variable substitution dropping something from the list) or by manipulating the changes output values as a final step in the changes job with a sed or whatever and updating the output with that outcome with another <code>run: echo "::set-output name=changes::blah,blarg"</code> which would replace the existing values.</p>
<p>Then the subsequent job conditional stays as-is, the exclude can be dropped from the matrix config.</p>
<p>You could also add the project path that you never want this workflow to run as part of the <code>on:</code> block:</p>
<pre><code class="lang-auto">on:
  push:
    paths-ignore:
      - 'something/path/**'
</code></pre>
<p>Then it just won’t ever trigger on changes to anything in that path. Which is way cleaner than triggering the workflow and running the changes job unnecessarily.</p>
